lib/webmention/parsers/html_parser.rb

```diff
@@ -0,0 +1,73 @@
+module Webmention
+  module Parsers
+    class HtmlParser < BaseParser
+      def self.mime_types
+        ['text/html']
+      end
+
+      Parsers.register(self)
+
+      HTML_ATTRIBUTE_MAP = {
+        cite:   %w[blockquote del ins q],
+        data:   %w[object],
+        href:   %w[a area],
+        poster: %w[video],
+        src:    %w[audio embed img source track video],
+        srcset: %w[img source]
+      }.freeze
+
+      CSS_SELECTORS_MAP = HTML_ATTRIBUTE_MAP.each_with_object({}) do |(attribute, elements), hash|
+        hash[attribute] = elements.map { |element| "#{element}[#{attribute}]" }
+      end
+
+      private
+
+      def doc
+        @doc ||= Nokogiri::HTML(response_body)
+      end
+
+      # Parse an HTML string for URLs
+      #
+      # @return [Array] the URLs
+      def parse_response_body
+        CSS_SELECTORS_MAP
+          .each_with_object([]) { |(*args), array| array << search_node(*args) }
+          .flatten
+          .map { |url| Absolutely.to_abs(base: response_url, relative: url) }
+          .uniq
+      end
```

Comment:
Each Parser implementation should define this `parse_response_body` private method and return an array of strings representing absolute URLs.